src/Tokenizer/Tokens.php

```diff
@@ -1139,6 +1157,26 @@ public function clearTokenAndMergeSurroundingWhitespace($index)
     }
 
     /**
+     * Finds the line number of the first token previous to the one at the specified index, including itself.
+     *
+     * @param int $index Token index
+     *
+     * @return int Line number
+     */
+    private function findNearestPrevTokenLineNumber($index)
```

Comment:
So, again, the outcome you wanna have and place you put the solution is 2 completely different things. Token is independent of runner. Keep things independent - if you want to have some data in some class - the data must be valid all the time. If it cannot be obtained, it proves you put it in wrong place.